.github/workflows/build.yml

```diff
@@ -17,135 +66,170 @@ jobs:
           - 'windows-latest'
           - 'macos-latest'
         python-version:
-          - '3.8'
-          - '3.9'
-          - '3.10'
-          - '3.11'
+          - '3.x'
+        node-version:
+          - '16.x'
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Need full history to determine version number.
+          fetch-depth: 0
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Cache node.js modules
+        uses: actions/cache@v3
+        with:
+          path: "**/node_modules"
+          key: ${{ runner.os }}-${{ matrix.node-version }}-node_modules-${{ hashFiles('**/package-lock.json') }}
+      - run: npm install
+      - name: Check for dirty working directory
+        run: git diff --exit-code
+      - name: Build wheel
+        run: pipx run nox -s dists
+      - name: Show package version
+        run: python setup.py --version
```

Comment:
I'm curious as to the intention here. I left it in because it doesn't prevent builds' success, but this step seems to add around 10-20 seconds to the CI runtime. Remember, the build jobs run concurrent to unit tests/docs jobs (as they no longer need the pkg installed), so it doesn't really matter if this is removed; I'm just curious.

Reply:
The intent was to be able to check the logs to verify that the setuptools-scm version generation was working as expected.  I'm a bit surprised that it is taking a long time.  Maybe a call to `git describe` by setuptools-scm is taking a long time for some reason?

Reply:
Thanks for the answer. The delay seems to the worst with macos and windows runners. It might something to do with the docker containers that github uses (trying to reference how the FS is exposed to the workflow).

In the bigger picture, I'm not worried about an extra 20 seconds when the docs take nearly 10 minutes to build with 3 different sphinx builders.

Reply:
I can have nox output the name of the wheel using `GITHUB_OUTPUT` env var. This should give us the version used without needing to rerun setup.py again.